contrib/babelfishpg_tsql/src/rolecmds.c

```diff
@@ -1580,6 +1663,22 @@ alter_bbf_authid_user_ext(AlterRoleStmt *stmt)
 
 	table_close(bbf_authid_user_ext_rel, RowExclusiveLock);
 
+	if (login_name_str)
+	{
+		if (old_login_name && strlen(old_login_name) > 0)
+		{
+			/* First revoke this user from old login as the user is being mapped to a new login. */
+			grant_revoke_role_to_login(old_login_name, stmt->role->rolename, NULL, false);
+			grant_revoke_role_to_login(old_login_name, stmt->role->rolename, "bbf_role_admin", false);
+			/* Now grant guest user to old login as it's mapped user is being removed. */
+			grant_revoke_role_to_login(old_login_name, get_guest_role_name(get_cur_db_name()), "bbf_role_admin", true);
+		}
+
+		/* Revoke guest user from new login as login now has a mapped user in current database. */
+		grant_revoke_role_to_login(login_name_str, get_guest_role_name(get_cur_db_name()), NULL, false);
+		grant_revoke_role_to_login(login_name_str, get_guest_role_name(get_cur_db_name()), "bbf_role_admin", false); /* revoke even membership granted by bbf_role_admin */
```

Comment:
Do we need to always revoke the guest membership? Even if older user was mapped to it?

Reply:
Yes, at this point this login will have a mapped user and guest user. Guest user is added as a member at the time of login creation.

Reply:
Where do we grant user to new login ?

Reply:
> Yes, at this point this login will have a mapped user and guest user. Guest user is added as a member at the time of login creation.

After this fix, It won't be needed to revoke guest every time, right? Because if any user is already mapped then login will be only user of that particular user. I was just suggesting to add a simple membership check and based on that we make the decision of executing revoke.

Reply:
We need to switch to sa user here as well plus lets also add tests here.

Let us also add a comment here explaining that older version did not store grantor information and after MVU to PG16 the grantor for these older role grants becomes `BOOTSTRAP_SUPERUSERID` in OSS.